content/docs/reference-glossary.md

```diff
@@ -0,0 +1,98 @@
+
+
+### [`props`](https://facebook.github.io/react/docs/components-and-props.html)
+`props` are inputs to a React component. They are data passed down from a parent component to a child component. `props` are readonly -- they should not be modified in any way. All React components must act like pure functions with respect to their `props`.
+
+### `this.props.children`
+`this.props.children` is available on every component. It contains the content between the opening and closing tags of a component. For example:
+
+```js
+<Welcome>Hello world!</Welcome>
+```
+The string `Hello world!` is available in `this.props.children` in the `Welcome` component.
+
+### [`state`](https://facebook.github.io/react/docs/state-and-lifecycle.html#adding-local-state-to-a-class)
+A component's `state` is a snapshot of the data contained in a component. `props` and `state` are different: `state` is user-defined, `props` are received from a parent component.
+
+## [Lifecycle Methods](https://facebook.github.io/react/docs/state-and-lifecycle.html#adding-lifecycle-methods-to-a-class)
+Lifecycle methods are custom functionality that gets executed during the different phases of a component. There are methods available when the component gets created and inserted into the DOM ([mounting](https://facebook.github.io/react/docs/react-component.html#mounting)), when the component updates and when the component gets unmounted or removed from the DOM. 
+
+ ## [Controlled](https://facebook.github.io/react/docs/forms.html#controlled-components) vs. [Uncontrolled Components](https://facebook.github.io/react/docs/uncontrolled-components.html)
+React has two different approaches to dealing with form inputs. An input form element whose value is controlled by React is called a *controlled component*. An *uncontrolled component* is an input that is just like any normal input we would use outside of React. When a user inputs data into a form field (an input box, dropdown, etc) the updated information is reflected. 
+
+## [Keys](https://facebook.github.io/react/docs/lists-and-keys.html) 
+ A "key" is a special string attribute you need to include when creating lists of elements. Keys help React identify which items have changed, are added, or are removed. Keys should be given to the elements inside an array to give the elements a stable identity.
+
+## [Refs](https://facebook.github.io/react/docs/refs-and-the-dom.html)
+React supports a special attribute that you can attach to any component. The `ref` attribute can be a string or a callback function. When the `ref` attribute is a callback function, the function receives the underlying DOM element as its argument. This allows you to have direct access to the DOM element.
```

Comment:
String refs are legacy and are likely to be removed in the future. I guess it's okay for us to mention them here though. 😄 

> the function receives the underlying DOM element as its argument

Maybe we should say "DOM element or component instance"?